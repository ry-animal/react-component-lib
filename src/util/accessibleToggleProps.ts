export function accessibleToggleProps(
  label: string,
  controls: string,
  expanded: boolean,
  popupType?: "menu"
) {
  return {
    "aria-controls": controls,
    "aria-expanded": expanded,
    "aria-haspopup": popupType || true,
    "aria-label": label,
    "aria-pressed": expanded,
    role: "button",
  };
}
