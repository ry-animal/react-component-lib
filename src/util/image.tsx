import { ISelectedFile } from "src/components/Forms/FileUpload/ISelectedFile";

export const makeDataUrl = (selectedFile: ISelectedFile) =>
  URL.createObjectURL(selectedFile.file);
