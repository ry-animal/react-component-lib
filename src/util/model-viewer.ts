/*
 * This file is a hacky way of allowing any other component to
 * use the <model-viewer /> tag just by doing:
 *   import "../../../util/model-viewer";
 *
 * The authors of the package do not expose it in a React-friendly
 * way, see: https://github.com/google/model-viewer/issues/1502
 *
 * The component documentation is at https://modelviewer.dev/docs/
 */

import "@google/model-viewer";

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace JSX {
    export interface IntrinsicElements {
      "model-viewer": {
        src?: string;
        alt?: string;
        poster?: string;
        autoplay?: boolean;
        exposure?: number;
        "camera-controls"?: boolean;
        "environment-image"?: string;
        "seamless-poster"?: string;
        loading?: "auto" | "lazy" | "eager";
        reveal?: "auto" | "interaction" | "manual";
      } & React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLElement>,
        HTMLElement
      >;
    }
  }
}
