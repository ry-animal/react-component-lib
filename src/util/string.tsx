const DEFAULT_LOCALE = "en-US";

export const truncateEthAddress = (str: string): string => {
  if (str.length) {
    const numStartingCharacters = 6;
    const numTrailingCharacters = 4;
    return `${str.substring(0, numStartingCharacters)}...${str.substring(
      str.length - numTrailingCharacters,
      str.length
    )}`;
  }
  return "";
};

export const formatUSDPriceString = (value: number): string => {
  const minimumFractionDigits = 2;
  const maximumFractionDigits = 2;
  return `$${value.toLocaleString(DEFAULT_LOCALE, {
    minimumFractionDigits,
    maximumFractionDigits,
  })}`;
};

export const formatETHPriceString = (priceInEth: number): string => {
  const minimumFractionDigits = 0;
  const maximumFractionDigits = 8;
  const maxDigitsWithDecimal = 8;

  let formatted = `${priceInEth
    .toLocaleString(DEFAULT_LOCALE, {
      minimumFractionDigits,
      maximumFractionDigits,
    })
    .replace(/,/gu, "")
    .substring(0, maxDigitsWithDecimal)}`;

  if (formatted.endsWith(".")) {
    formatted = formatted.slice(0, -1);
  }
  return formatted;
};

const evaluateNumberLessThanOne = (num: number) => {
  if (num === 0) {
    return "0";
  }

  const minDecimalValue = 0.0001;
  if (num < minDecimalValue) {
    return minDecimalValue.toString();
  }
  // To round decimal to 4 digits
  const maxDigits = 10000;
  return `${Math.round(num * maxDigits) / maxDigits}`;
};

// Abbreviate Number https://stackoverflow.com/a/9462382

export const abbreviateNumber = (num: number): string => {
  const lookup = [
    { value: 1, symbol: "" },
    { value: 1e3, symbol: "k" },
    { value: 1e6, symbol: "M" },
    { value: 1e9, symbol: "B" },
    { value: 1e12, symbol: "T" },
    { value: 1e15, symbol: "P" },
    { value: 1e18, symbol: "E" },
  ];

  const rx = /\.0+$|(?<id>\.[0-9]*[1-9])0+$/u;
  const digits = 1;

  const item = lookup
    .slice()
    .reverse()
    .find((x) => num >= x.value);
  return item
    ? (num / item.value).toFixed(digits).replace(rx, "$1") + item.symbol
    : evaluateNumberLessThanOne(num);
};

export const getDomainFromRegex = (url: string) => {
  const matches = url.match(/:\/\/(?<id>.[^/]+)/u);
  // Extract domain (will be null if no match is found)
  return matches && matches[1];
};
