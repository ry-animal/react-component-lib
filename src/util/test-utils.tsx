import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
