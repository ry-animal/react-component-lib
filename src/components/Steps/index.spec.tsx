import { Step, Steps } from ".";
import { render } from "../../util/test-utils";

describe("<Steps />", () => {
  const steps: [Step, Step, ...Step[]] = [
    { label: "1", description: "open the fridge" },
    { label: "2", description: "take ice cream out" },
    { label: "3", description: "eat the ice cream" },
  ];
  const stepsProps = {
    steps,
    currentStep: 0,
  };

  describe("rendering", () => {
    test("renders horizontal 3 steps by default", () => {
      expect(render(<Steps {...stepsProps} />).asFragment()).toMatchSnapshot();
    });

    test("horizontal render container should have flex-direction row and align-items center", () => {
      expect(
        render(<Steps {...stepsProps} orientation="horizontal" />).asFragment()
      ).toMatchSnapshot();
    });

    test("vertical render container should have flex-direction column and align-items flex-start", () => {
      expect(
        render(<Steps {...stepsProps} orientation="vertical" />).asFragment()
      ).toMatchSnapshot();
    });
  });
});
