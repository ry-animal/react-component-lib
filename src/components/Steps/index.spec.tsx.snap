// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Steps /> rendering horizontal render container should have flex-direction row and align-items center 1`] = `
<DocumentFragment>
  .c3 {
  text-align: center;
  width: var(--circleDiam);
  height: var(--circleDiam);
  font-weight: 500;
  font-size: calc(var(--circleDiam) / 2);
  color: #F9F9F9;
  background: #151515;
  border: 0.1em solid #151515;
  border-radius: 50%;
  line-height: var(--circleDiam);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  text-align: center;
  width: var(--circleDiam);
  height: var(--circleDiam);
  font-weight: 500;
  font-size: calc(var(--circleDiam) / 2);
  color: #F9F9F9;
  background: #A4A4A4;
  border: 0.1em solid #A4A4A4;
  border-radius: 50%;
  line-height: var(--circleDiam);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  text-align: center;
  color: #151515;
  font-size: calc(var(--circleDiam) / 2);
  padding: 0.25em;
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  --lineWidth: 4px;
  --circleDiam: 2rem;
  --verticalPaddingBottom: 4rem;
  --verticalGap: 2rem;
  --verticalBottom: 1rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #151515;
  text-align: center;
}

.c1:before {
  position: relative;
  z-index: 1;
  display: block;
}

.c1:first-child {
  padding-top: var(--lineWidth);
}

.c1:not(:first-child):before {
  content: "";
  background-color: #151515;
  width: calc(100% - var(--circleDiam));
  height: var(--lineWidth);
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  right: calc(50% - var(--circleDiam) / 2 - 0.25%);
  top: 1.25rem;
}

.c5 {
  --lineWidth: 4px;
  --circleDiam: 2rem;
  --verticalPaddingBottom: 4rem;
  --verticalGap: 2rem;
  --verticalBottom: 1rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #A4A4A4;
  text-align: center;
}

.c5:before {
  position: relative;
  z-index: 1;
  display: block;
}

.c5:first-child {
  padding-top: var(--lineWidth);
}

.c5:not(:first-child):before {
  content: "";
  background-color: #A4A4A4;
  width: calc(100% - var(--circleDiam));
  height: var(--lineWidth);
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  right: calc(50% - var(--circleDiam) / 2 - 0.25%);
  top: 1.25rem;
}

.c0 {
  font-family: 'Poppins',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

<div
    class="c0"
    orientation="horizontal"
  >
    <div
      class="c1"
      orientation="horizontal"
    >
      <div
        class="c2"
        orientation="horizontal"
      >
        <span
          class="c3"
        >
          1
        </span>
        <span
          class="c4"
        >
          open the fridge
        </span>
      </div>
    </div>
    <div
      class="c5"
      orientation="horizontal"
    >
      <div
        class="c2"
        orientation="horizontal"
      >
        <span
          class="c6"
        >
          2
        </span>
        <span
          class="c4"
        >
          take ice cream out
        </span>
      </div>
    </div>
    <div
      class="c5"
      orientation="horizontal"
    >
      <div
        class="c2"
        orientation="horizontal"
      >
        <span
          class="c6"
        >
          3
        </span>
        <span
          class="c4"
        >
          eat the ice cream
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Steps /> rendering renders horizontal 3 steps by default 1`] = `
<DocumentFragment>
  .c3 {
  text-align: center;
  width: var(--circleDiam);
  height: var(--circleDiam);
  font-weight: 500;
  font-size: calc(var(--circleDiam) / 2);
  color: #F9F9F9;
  background: #151515;
  border: 0.1em solid #151515;
  border-radius: 50%;
  line-height: var(--circleDiam);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  text-align: center;
  width: var(--circleDiam);
  height: var(--circleDiam);
  font-weight: 500;
  font-size: calc(var(--circleDiam) / 2);
  color: #F9F9F9;
  background: #A4A4A4;
  border: 0.1em solid #A4A4A4;
  border-radius: 50%;
  line-height: var(--circleDiam);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  text-align: center;
  color: #151515;
  font-size: calc(var(--circleDiam) / 2);
  padding: 0.25em;
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  --lineWidth: 4px;
  --circleDiam: 2rem;
  --verticalPaddingBottom: 4rem;
  --verticalGap: 2rem;
  --verticalBottom: 1rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #151515;
  text-align: center;
}

.c1:before {
  position: relative;
  z-index: 1;
  display: block;
}

.c1:first-child {
  padding-top: var(--lineWidth);
}

.c1:not(:first-child):before {
  content: "";
  background-color: #151515;
  width: calc(100% - var(--circleDiam));
  height: var(--lineWidth);
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  right: calc(50% - var(--circleDiam) / 2 - 0.25%);
  top: 1.25rem;
}

.c5 {
  --lineWidth: 4px;
  --circleDiam: 2rem;
  --verticalPaddingBottom: 4rem;
  --verticalGap: 2rem;
  --verticalBottom: 1rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #A4A4A4;
  text-align: center;
}

.c5:before {
  position: relative;
  z-index: 1;
  display: block;
}

.c5:first-child {
  padding-top: var(--lineWidth);
}

.c5:not(:first-child):before {
  content: "";
  background-color: #A4A4A4;
  width: calc(100% - var(--circleDiam));
  height: var(--lineWidth);
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  right: calc(50% - var(--circleDiam) / 2 - 0.25%);
  top: 1.25rem;
}

.c0 {
  font-family: 'Poppins',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

<div
    class="c0"
    orientation="horizontal"
  >
    <div
      class="c1"
      orientation="horizontal"
    >
      <div
        class="c2"
        orientation="horizontal"
      >
        <span
          class="c3"
        >
          1
        </span>
        <span
          class="c4"
        >
          open the fridge
        </span>
      </div>
    </div>
    <div
      class="c5"
      orientation="horizontal"
    >
      <div
        class="c2"
        orientation="horizontal"
      >
        <span
          class="c6"
        >
          2
        </span>
        <span
          class="c4"
        >
          take ice cream out
        </span>
      </div>
    </div>
    <div
      class="c5"
      orientation="horizontal"
    >
      <div
        class="c2"
        orientation="horizontal"
      >
        <span
          class="c6"
        >
          3
        </span>
        <span
          class="c4"
        >
          eat the ice cream
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Steps /> rendering vertical render container should have flex-direction column and align-items flex-start 1`] = `
<DocumentFragment>
  .c3 {
  text-align: center;
  width: var(--circleDiam);
  height: var(--circleDiam);
  font-weight: 500;
  font-size: calc(var(--circleDiam) / 2);
  color: #F9F9F9;
  background: #151515;
  border: 0.1em solid #151515;
  border-radius: 50%;
  line-height: var(--circleDiam);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  text-align: center;
  width: var(--circleDiam);
  height: var(--circleDiam);
  font-weight: 500;
  font-size: calc(var(--circleDiam) / 2);
  color: #F9F9F9;
  background: #A4A4A4;
  border: 0.1em solid #A4A4A4;
  border-radius: 50%;
  line-height: var(--circleDiam);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  text-align: center;
  color: #151515;
  font-size: calc(var(--circleDiam) / 2);
  padding: 0.25em;
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  --lineWidth: 4px;
  --circleDiam: 2rem;
  --verticalPaddingBottom: 4rem;
  --verticalGap: 2rem;
  --verticalBottom: 1rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #151515;
  text-align: center;
}

.c1:before {
  position: relative;
  z-index: 1;
  display: block;
  height: calc(var(--verticalPaddingBottom) + var(--verticalBottom) * 2);
}

.c1:first-child {
  padding-top: var(--lineWidth);
}

.c1:not(:first-child):before {
  content: "";
  background-color: #151515;
  width: var(--lineWidth);
  left: calc(var(--circleDiam) / 2);
  height: calc(var(--verticalPaddingBottom) + var(--verticalBottom) * 2);
}

.c5 {
  --lineWidth: 4px;
  --circleDiam: 2rem;
  --verticalPaddingBottom: 4rem;
  --verticalGap: 2rem;
  --verticalBottom: 1rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #A4A4A4;
  text-align: center;
}

.c5:before {
  position: relative;
  z-index: 1;
  display: block;
  height: calc(var(--verticalPaddingBottom) + var(--verticalBottom) * 2);
}

.c5:first-child {
  padding-top: var(--lineWidth);
}

.c5:not(:first-child):before {
  content: "";
  background-color: #A4A4A4;
  width: var(--lineWidth);
  left: calc(var(--circleDiam) / 2);
  height: calc(var(--verticalPaddingBottom) + var(--verticalBottom) * 2);
}

.c0 {
  font-family: 'Poppins',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: 100%;
  height: 100%;
}

<div
    class="c0"
    orientation="vertical"
  >
    <div
      class="c1"
      orientation="vertical"
    >
      <div
        class="c2"
        orientation="vertical"
      >
        <span
          class="c3"
        >
          1
        </span>
        <span
          class="c4"
        >
          open the fridge
        </span>
      </div>
    </div>
    <div
      class="c5"
      orientation="vertical"
    >
      <div
        class="c2"
        orientation="vertical"
      >
        <span
          class="c6"
        >
          2
        </span>
        <span
          class="c4"
        >
          take ice cream out
        </span>
      </div>
    </div>
    <div
      class="c5"
      orientation="vertical"
    >
      <div
        class="c2"
        orientation="vertical"
      >
        <span
          class="c6"
        >
          3
        </span>
        <span
          class="c4"
        >
          eat the ice cream
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
